app/routers/api/files.py

```
@@ -128,7 +128,137 @@ def _serialize_file(
+    if max_size is not None:
+        query = query.filter(FileInventory.file_size <= max_size)
+    if min_mtime is not None:
+        query = query.filter(FileInventory.file_mtime >= min_mtime)
+    if max_mtime is not None:
+        query = query.filter(FileInventory.file_mtime <= max_mtime)
+    if storage_location_id is not None:
+        query = query.filter(FileInventory.cold_storage_location_id == storage_location_id)
+    return query
+
+
+def _apply_cursor_pagination(query, sort_field, cursor_data, is_descending):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_apply_cursor_pagination`

(ANN202)